core/tests/protractor_utils/users.js

```diff
@@ -102,9 +118,22 @@ var _completeSignup = async function(username) {
   await action.click('Agree to terms checkbox', agreeToTermsCheckbox);
 
   var registerUser = element(by.css('.protractor-test-register-user'));
+
+  // Signup page performs client side redirection which is known to cause
+  // "both angularJS testability and angular testability are undefined" flake.
+  // As suggested protractor devs here (http://git.io/v4gXM), waiting for
+  // angular is disabled during client side redirects.
+  await browser.waitForAngularEnabled(false);
   await action.click('Register user button', registerUser);
 
+  // The await action.click only waits until button is click, it doesnot wait
+  // for the redirection to trigger. So, manually waiting for redirection here.
+  // eslint-disable-next-line oppia/protractor-practices
+  await browser.sleep(3000);
```

Comment:
Is there any other way to wait for redirection? Can we wait for the invisibility/visibility of some element on the page in order to detect the redirection?

Reply:
changed.